src/Illuminate/Support/Testing/Fakes/QueueFake.php

```diff
@@ -6,20 +6,44 @@
 use Closure;
 use Illuminate\Contracts\Queue\Queue;
 use Illuminate\Queue\QueueManager;
+use Illuminate\Support\Collection;
 use Illuminate\Support\Traits\ReflectsClosures;
 use PHPUnit\Framework\Assert as PHPUnit;
 
 class QueueFake extends QueueManager implements Queue
 {
     use ReflectsClosures;
 
+    /**
+     * The original queue manager.
+     *
+     * @var \Illuminate\Contracts\Queue\Queue
+     */
+    protected $queue;
+
+    /**
+     * The job types that should be intercepted instead of pushed to the queue.
+     *
+     * @var array
+     */
+    protected $jobsToFake;
+
     /**
      * All of the jobs that have been pushed.
      *
      * @var array
      */
     protected $jobs = [];
 
+    public function __construct($app, QueueManager $queue, $jobsToFake = [])
```

Comment:
This is a major breaking change as you're adding a new dependency to this class.

Reply:
Should I send it to master?